src/Prototype/Runtime/Errors.hs

```diff
@@ -0,0 +1,47 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE GADTs #-}
+module Prototype.Runtime.Errors
+  ( RuntimeErr(..)
+  , IsRuntimeErr(..)
+  ) where
+
+import qualified Network.HTTP.Types            as HTTP
+
+-- | A generalised error
+data RuntimeErr where
+  -- | Capture known error types. 
+  KnownErr ::IsRuntimeErr e => e -> RuntimeErr
```

Comment:
Is the `::` without separating space on purpose or a typo ?

Reply:
This was formatted using Brittany; which currently has a bug with GADT formatting; https://github.com/lspitzner/brittany/issues/217

I've been meaning to take a crack at it myself.